File: GeneratedEntityReaderClasses/Classes/types_aabb.cs
namespace GeneratedNoitaClasses;
public class types_aabb :  iNoitaType<types_aabb>{
	public NoitaCVector2<NoitaFloat> min { get; set; } = new();
	public NoitaCVector2<NoitaFloat> max { get; set; } = new();

    public void Read(NoitaStream s){
		min.Read(s);
		max.Read(s);

    }
    public void Write(NoitaStream s){
		min.Write(s);
		max.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/types_aabb.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaCVector2
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class types_aabb:
    min:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaFloat))
    max:NoitaCVector2 = field(default_factory=lambda: NoitaCVector2(NoitaFloat))
    
    def Read(self, s:NoitaStream):
        self.min.Read(s)
        self.max.Read(s)
    
    def Write(self, s:NoitaStream):
        self.min.Write(s)
        self.max.Write(s)

